dockers/docker-dhcp-relay/docker-dhcp-relay.supervisord.conf.j2

```diff
@@ -38,31 +38,73 @@ stderr_logfile=syslog
 dependent_startup=true
 dependent_startup_wait_for=rsyslogd:running
 
+{% set num_relays = { 'count': 0 } %}
 {# If our configuration has VLANs... #}
 {% if VLAN %}
 {# Count how many VLANs require a DHCP relay agent... #}
-{% set num_relays = { 'count': 0 } %}
 {% for vlan_name in VLAN %}
 {% if VLAN[vlan_name]['dhcp_servers'] %}
 {% set _dummy = num_relays.update({'count': num_relays.count + 1}) %}
 {% endif %}
 {% endfor %}
-{# If one or more of the VLANs require a DHCP relay agent... #}
+{% endif %}
+{# If our configuration has Portchannels... #}
+{% if PORTCHANNEL_INTERFACE %}
+{# Count how many Portchannels interfaces require a DHCP relay agent... #}
+{% for pc_name in PORTCHANNEL_INTERFACE %}
+{% if PORTCHANNEL_INTERFACE[pc_name]['dhcp_servers'] %}
+{% set _dummy = num_relays.update({'count': num_relays.count + 1}) %}
+{% endif %}
+{% endfor %}
+{% endif %}
+{# If our configuration has Ethernet.. #}
+{% if INTERFACE %}
+{# Count how many Ethernet interfaces require a DHCP relay agent... #}
+{% for intf_name in INTERFACE %}
+{% if INTERFACE[intf_name]['dhcp_servers'] %}
+{% set _dummy = num_relays.update({'count': num_relays.count + 1}) %}
+{% endif %}
+{% endfor %}
+{% endif %}
+
+{# If one or more of the VLANs/Portchannel/Ethernet interfaces require a DHCP relay agent... #}
 {% if num_relays.count > 0 %}
 [group:isc-dhcp-relay]
 programs=
 {%- set add_preceding_comma = { 'flag': False } %}
+{% if VLAN %}
 {% for vlan_name in VLAN %}
 {% if VLAN[vlan_name]['dhcp_servers'] %}
 {% if add_preceding_comma.flag %},{% endif %}
 {% set _dummy = add_preceding_comma.update({'flag': True}) %}
 isc-dhcp-relay-{{ vlan_name }}
 {%- endif %}
 {% endfor %}
-
+{% endif %}
+{% if PORTCHANNEL_INTERFACE %}
+{% for pc_name in PORTCHANNEL_INTERFACE %}
+{% if PORTCHANNEL_INTERFACE[pc_name]['dhcp_servers'] %}
+{% if add_preceding_comma.flag %},{% endif %}
+{% set _dummy = add_preceding_comma.update({'flag': True}) %}
+isc-dhcp-relay-{{ pc_name }}
```

Comment:
Why are we attaching a relay agent to the port channels and non-vlan interfaces?

Reply:
DHCP relay agent should be supported on all L3 interfaces right.

For an MCLAG deployment,
- DHCP server must be reachable from both the MC-LAG peers.
- Clients may send the DHCP DISCOVER packet to either of the MC-LAG peers.

Reply:
@AkhileshSamineni It is not clear to me still why on MCLAG setup a relay is required on L3 interfaces. What nodes are we bootstrapping here? For compute nodes (server) that are connected to T0, DHCP relay on T0 vlan interfaces do capture those packet and relay them to DHCP server. Also DHCP relay agent should not relay a packet that was already relayed. So, are we bootstrapping router/switches?

Reply:
@tahmed-dev For completeness, depending on different deployments usecases, users can enable dhcp-relay on different types of L3 interfaces. So extending dhcp-relay support on other L3 interfaces.

Reply:
@AkhileshSamineni Thanks. If this is the case, it is better to control such DHCP instances using feature flag such that they are not started on every L# port for all deployments. This will have CPU impact as I could see some redundant activation of different DHCP instances when DORA packets travel through the mgmt plane.

Reply:
Added a new command to enable/disable such DHCP instances.
Please check the PR for more details - https://github.com/Azure/sonic-utilities/pull/1311